openmdao/core/system.py

```diff
@@ -4230,6 +4239,36 @@ def get_io_metadata(self, iotypes=('input', 'output'), metadata_keys=None,
                             ret_meta = None
 
                 if ret_meta is not None:
+                    # handle is_indep_var
+                    if is_indep_var is not None:
+                        if iotype == 'output':
+                            out_meta = meta
+                        else:
+                            src_name = self.get_source(abs_name)
+                            try:
+                                out_meta = metadict['output'][src_name]
+                            except KeyError:
+                                out_meta = disc_metadict['output'][src_name]
+
+                        src_tags = out_meta['tags'] if 'tags' in out_meta else {}
+                        if is_indep_var is True and 'openmdao:indep_var' not in src_tags:
+                            continue
+                        elif is_indep_var is False and 'openmdao:indep_var' in src_tags:
+                            continue
+
+                    # handle is_design_var
+                    if is_design_var is not None:
+                        des_vars = self.get_design_vars(use_prom_ivc=False)
```

Comment:
We should probably move this outside of the loop.  Also, since we don't need sizing info, you could pass `get_sizes=False` into the call as well.

Reply:
oops.. I thought I had done this, but apparently only did it in Case.   wilco on get_sizes.